### ISSUE TITLE
[a11y]: The combobox 'aria-expanded' attribute is false, but the combobox popup is visible

### ISSUE THREAD

**Issue by pnayek** [carbon-design-system/carbon#16093]
### Package

carbon-components, carbon-components-react

### Browser

Chrome

### Operating System

MacOS

### Package version

"carbon-components": "10.58.13", "carbon-components-react": "7.59.19"

### Automated testing tool and ruleset

IBM Accessibility Checker tool - The combobox 'aria-expanded' attribute is false, but the combobox popup is visible

### Description

Hi Team,
The IBM Accessibility Checker tool reports this issue: `The combobox 'aria-expanded' attribute is false, but the combobox popup is visible` when we are using a `Dropdown` component. 

- Is this a known Carbon issue? We are using "carbon-components": "10.58.13", "carbon-components-react": "7.59.19".
- Is there a best practice that needs to be followed to get past this issue?

<img width="1705" alt="image" src="https://github.com/carbon-design-system/carbon/assets/15056711/5fb0c56b-042e-48fd-8d9e-56398e4caa78">

Below is how we have written the dropdown component:
```
<Dropdown
					id={"some-dropdown"}
					className={somestyle....}
					label={"some label..."}
					items={items}
					type="inline"
					onChange={(evt) => this._onSelectAssetDropdown(evt)}
					initialSelectedItem={a state...}
					direction="bottom"
					size="lg"
					titleText={the title...}
					hideLabel
				/>
```

### WCAG 2.1 Violation

4.1.2 Name, Role, Value

### Reproduction/example

N/A

### Steps to reproduce

N/A

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tay1orjones** [comment]
Hi, could you reproduce the issue in a [stackblitz](https://stackblitz.com/github/carbon-design-system/sandboxes/tree/main/react/latest/react-18-vite/?file=src%2Findex.scss,src%2FApp.jsx&preset=node=), and post the link back here? It's hard to know exactly what's going wrong without seeing the issue firsthand.

**Comment by tay1orjones** [comment]
Actually, I can see this error in the [v10 storybook](https://v7-react.carbondesignsystem.com/?path=/story/components-dropdown--default)

This error was not present back in [v10.55.0](https://deploy-preview-10955--v11-carbon-react.netlify.app/?path=/story/components-dropdown--default). Between then and the current v10.59.21 release there isn't much that has changed in Dropdown other than the `role="combobox"` being added to the element. So my guess is https://github.com/carbon-design-system/carbon/pull/15197 technically caused this regression.

That said, all of this is fixed in v11. It uses the latest version of Downshift.js which is 3 major versions ahead of the Downshift version used in v10.

**Comment by pnayek** [comment]
Thanks for looking into that, @tay1orjones! We will be migrating to Carbon11 very soon, and I hope that will help resolve the issue.

**Comment by keithcha** [comment]
Would it be too much to ask for this to be ported to v10?
